modules/040-node-manager/templates/node-group/_bootstrap.tpl

```diff
@@ -55,39 +41,153 @@ EOF
+      rm -rf "${TMP_DIR}" "${BB_FETCHED_PACKAGES_STORE:?}/${PACKAGE}"
+    done
+  }
+
+  bb-package-fetch() {
+    mkdir -p "${BB_FETCHED_PACKAGES_STORE}"
+    declare -A PACKAGES_MAP
+    local PACKAGE_WITH_DIGEST
+    for PACKAGE_WITH_DIGEST in "$@"; do
+      local PACKAGE=""
+      local DIGEST=""
+      PACKAGE="$(awk -F ":" '{print $1}' <<< "${PACKAGE_WITH_DIGEST}")"
+      DIGEST="$(awk -F ":" '{print $2":"$3}' <<< "${PACKAGE_WITH_DIGEST}")"
+      PACKAGES_MAP[$DIGEST]="${PACKAGE}"
+    done
+    bb-package-fetch-blobs PACKAGES_MAP
+  }
+
+  bb-package-fetch-blobs() {
+    local PACKAGE_DIGEST
+    for PACKAGE_DIGEST in "${!PACKAGES_MAP[@]}"; do
+      local PACKAGE_DIR="${BB_FETCHED_PACKAGES_STORE}/${PACKAGES_MAP[$PACKAGE_DIGEST]}"
+      mkdir -p "${PACKAGE_DIR}"
+      bb-package-fetch-blob "${PACKAGE_DIGEST}" "${PACKAGE_DIR}/${PACKAGE_DIGEST}.tar.gz"
+    done
+  }
+
+  bb-package-fetch-blob() {
+    check_python
+
+    cat - <<EOFILE | $python_binary
+import random
+import ssl
+try:
+  from urllib.request import urlopen, Request, HTTPError
+except ImportError:
+  from urllib2 import urlopen, Request, HTTPError
+endpoints = "${PACKAGES_PROXY_ADDRESSES}".split(",")
+# Choose a random endpoint as first ep, that increase fault tolerance and reduce load on first endpoint.
+endpoints.insert(0, random.choice(endpoints))
+ssl._create_default_https_context = ssl._create_unverified_context
+for ep in endpoints:
+  url = 'https://{}/package?digest=$1&repository=${REPOSITORY}'.format(ep)
+  request = Request(url, headers={'Authorization': 'Bearer ${PACKAGES_PROXY_TOKEN}'})
+  try:
+    response = urlopen(request, timeout=300)
+  except HTTPError as e:
+    print("Access to {} return HTTP Error {}: {}".format(url, e.getcode(), e.read()[:255]))
+    continue
+  break
+with open('$2', 'wb') as f:
+  f.write(response.read())
+EOFILE
+  }
+  {{- with $context.Values.global.modulesImages.digests.registrypackages }}
+  bb-package-install "jq:{{ .jq16 }}" "curl:{{ .d8Curl821 }}" "netcat:{{ .netcat110481 }}"
+    {{- if hasKey $context.Values.nodeManager.internal "cloudProvider" }}
+      {{- if eq $context.Values.nodeManager.internal.cloudProvider.type "aws" }}
+  bb-package-install "ec2DescribeTags:{{ .ec2DescribeTagsV001Flant2 }}" 
+      {{- end }}
+    {{- end }}
+  {{- end }}
+}
+
   {{- if not (hasKey $ng "staticInstances") }}
+    {{- if hasKey $context.Values.nodeManager.internal "cloudProvider" }}
 function run_cloud_network_setup() {
-  {{- if hasKey $context.Values.nodeManager.internal "cloudProvider" }}
-    {{- if $bootstrap_script_common := $context.Files.Get (printf "candi/cloud-providers/%s/bashible/common-steps/bootstrap-networks.sh.tpl" $context.Values.nodeManager.internal.cloudProvider.type)  }}
+      {{- if $bootstrap_script_common := $context.Files.Get "candi/bashible/bootstrap-networks.sh.tpl" }}
   cat > ${BOOTSTRAP_DIR}/cloud-provider-bootstrap-networks.sh <<"EOF"
```

Comment:
Why we need cat here ? Include template into the function

Reply:
I did not dare to delete this, because I suspected that there were some specific cases where they were needed. I’ll try to check the work without them.